packages/form-js-editor/src/features/properties-panel/groups/ValidationGroup.js

```diff
@@ -265,25 +266,27 @@ function Max(props) {
 function ValidationType(props) {
   const {
     field,
+    editField,
     getValue,
-    id,
-    onChange
+    id
   } = props;
 
   const debounce = useService('debounce');
 
-  const clearCustomValidation = () => {
-    onChange('minLength')(undefined);
-    onChange('maxLength')(undefined);
-    onChange('pattern')(undefined);
-  };
-
   const setValue = (validationType) => {
+
+    let validate = get(field, [ 'validate' ], {});
+
     if (validationType) {
-      clearCustomValidation();
+      delete validate.minLength;
```

Comment:
This mutates the form field before the command is executed breaking undo. 😢

Reply:
Makes a lot of sense, gotta clone the object first.

Reply:
The new value of `validation` must be a new object. The old value must remain untouched.

Reply:
Yeah I've already gotten this into this PR via the spread operator. We just haven't really got any test cases that are similar within form-js for this. 

I think you may be right and this should be implemented as a behavior rather than within the properties panel. Because this functionality shouldn't only run when we use the properties panel, but also programmatically.